src/utils/index.js

```diff
@@ -30,16 +32,154 @@ export function loadScript(scriptName, cb) {
     s.parentNode.insertBefore(script, s.nextSibling);
 }
 
+function isObject(value) {
+    const type = typeof value;
+    return value != null && (type == 'object' || type == 'function');
+}
 
-export function debounce(callback, wait, context = this) {
-    let timeout = null;
-    let callbackArgs = null;
+export function debounce(func, wait, options) {
```

Comment:
instead of copy pasting the implementation, just use the npm package: https://www.npmjs.com/package/lodash.debounce